code/modules/mob/living/basic/space_fauna/demon/demon_subtypes.dm

```diff
@@ -0,0 +1,163 @@
+	bare_wound_bonus = initial(bare_wound_bonus)
+
+/// Performs the classic slaughter demon bodyslam on the attack_target. Yeets them a screen away.
+/mob/living/basic/demon/slaughter/proc/bodyslam(atom/attack_target)
+	if(!isliving(attack_target))
+		return
+
+	if(!Adjacent(attack_target))
+		to_chat(src, span_warning("You are too far away to use your slam attack on [attack_target]!"))
+		return
+
+	if(!COOLDOWN_FINISHED(src, slam_cooldown))
+		to_chat(src, span_warning("Your slam ability is still on cooldown!"))
+		return
+
+	face_atom(attack_target)
+
+	var/mob/living/victim = attack_target
+	victim.take_bodypart_damage(brute = 20, wound_bonus = wound_bonus) // don't worry, there's more punishment when they hit something
+
+	visible_message(
+		span_danger("[src] slams into [victim] with monstrous strength!"),
+		span_danger("You slam into [victim] with monstrous strength!"),
+		ignored_mobs = victim,
+	)
+	to_chat(victim, span_userdanger("[src] slams into you with monstrous strength, sending you flying like a ragdoll!"))
+
+	var/turf/yeet_target = get_edge_target_turf(victim, dir)
+	victim.throw_at(yeet_target, 10, 5, src)
+	COOLDOWN_START(src, slam_cooldown, slam_cooldown_time)
+	log_combat(src, victim, "slaughter slammed")
+
+/// Proc that we execute on attacking someone to keep track of our hitstreaks and wound bonuses. Also handles triggering the bodyslam on attacks.
+/mob/living/basic/demon/slaughter/proc/on_attack(mob/living/source, atom/attack_target, proximity_flag, list/modifiers)
+	SIGNAL_HANDLER
+
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		bodyslam(attack_target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!iscarbon(attack_target))
+		return
+
+	var/mob/living/carbon/target = attack_target
+	if(target.stat == DEAD || isnull(target.mind) || (current_hitstreak > wound_bonus_hitstreak_max))
+		return
+
+	current_hitstreak++
+	wound_bonus += wound_bonus_per_hit
+	bare_wound_bonus += wound_bonus_per_hit
+
+/// The laughter demon! It's everyone's best friend! It just wants to hug them so much, it wants to hug everyone at once!
+/mob/living/basic/demon/slaughter/laughter
+	name = "laughter demon"
+	real_name = "laughter demon"
+	desc = "A large, adorable creature covered in armor with pink bows."
+	speak_emote = list("giggles", "titters", "chuckles", "guffaws", "laughs")
+	response_help_continuous = "hugs"
+	attack_verb_continuous = "wildly tickles"
+	attack_verb_simple = "wildly tickle"
+
+	attack_sound = 'sound/items/bikehorn.ogg'
+	attack_vis_effect = null
+	death_sound = 'sound/misc/sadtrombone.ogg'
+
+	icon_state = "bow_demon"
+	icon_living = "bow_demon"
+	death_message = "fades out, as all of its friends are released from its prison of hugs."
+
+	antag_type = /datum/antagonist/slaughter/laughter
+	crawl_type = /datum/action/cooldown/spell/jaunt/bloodcrawl/slaughter_demon/funny
+
+/mob/living/basic/demon/slaughter/laughter/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_LIVING_DEATH, PROC_REF(on_death))
```

Comment:
Why is this not just extending `/death()`, there's really no problem with it

Reply:
because i want it to execute at the same time all of the corpses are being ejected out of the laughter demon which listens to the same signal on the datum action cooldown. i know the difference in practicality would be a handful of milliseconds but it flows better for my mind that when all of the mobs are ejected/spawned on the recieval of `COMSIG_LIVING_DEATH`, that we also spawn the cat on `COMSIG_LIVING_DEATH` since it's all tied to that same "step" on the signal, rather than be done on the subtype. `COMSIG_LIVING_DEATH` handling both the corpse-ejection and the spawning of the kitten intuitively makes sense to me at that point in the proc.

if we didn't have any other functionality on the laughter demon that didn't leverage `COMSIG_LIVING_DEATH`, it would be fine to extend it then imo, but I much rather would have both things tie together in that way

Reply:
![image](https://github.com/tgstation/tgstation/assets/34697715/0e94228f-c529-4281-98ed-80f65ffb9ebd)

also because i want the kitten to spawn while the demon is on screen and spewing out all of the corpses, because we `DEL_ON_DEATH` on the basic version of `death()` (and the signal is sent earlier in parent)